prov/util/src/util_domain.c

```diff
@@ -64,8 +64,9 @@ static int util_domain_init(struct util_domain *domain,
 {
 	atomic_initialize(&domain->ref, 0);
 	fastlock_init(&domain->lock);
-	domain->caps = info->caps;
-	domain->mode = info->mode;
+	domain->caps = info->caps | info->domain_attr->caps;
+	domain->mode = info->mode | info->domain_attr->mode;
```

Comment:
Now that we have domain specific mode/caps, I think setting domain->caps/mode will lead to confusion later.  Maybe rename these to all_caps, info_caps, merged_caps, naming_is_hard_caps to better indicate that these are not the domain specific flags.